File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/kit/KitService.java
package top.likoslupus.cellulosesz.api.kit;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface KitService {

    CompletableFuture<Void> reload();

    List<KitDefinition> kits();

    Optional<KitDefinition> kit(String id);

    CompletableFuture<Void> save(KitDefinition kit);

    CompletableFuture<Boolean> delete(String id);

    CompletableFuture<KitClaimResult> claim(CellPlayer player, KitDefinition kit);

    CompletableFuture<Void> resetCooldown(UUID uuid, String kitId);

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/kit/KitService.kt
package top.likoslupus.cellulosesz.api.kit

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.util.*
import java.util.concurrent.CompletableFuture

public interface KitService {

    public fun kits(): List<KitDefinition>

    public fun kit(id: String): KitDefinition?

    public fun save(kit: KitDefinition): CompletableFuture<Void>

    public fun delete(id: String): CompletableFuture<Boolean>

    public fun claim(player: CellPlayer, kit: KitDefinition): CompletableFuture<KitClaimResult>

    public fun resetCooldown(uuid: UUID, kitId: String): CompletableFuture<Void>

}
